source/index.js

```diff
@@ -1,751 +1,403 @@
 
-        if (geometry.normals.length > 0) {
+        if ( geometry.normals.length > 0 ) {
 
-          buffergeometry.addAttribute('normal', new THREE.BufferAttribute(new Float32Array(geometry.normals), 3));
+          buffergeometry.addAttribute( 'normal', new THREE.BufferAttribute( new Float32Array( geometry.normals ), 3 ) );
 
         } else {
 
           buffergeometry.computeVertexNormals();
 
         }
 
-        if (geometry.uvs.length > 0) {
+        if ( geometry.uvs.length > 0 ) {
 
-          buffergeometry.addAttribute('uv', new THREE.BufferAttribute(new Float32Array(geometry.uvs), 2));
+          buffergeometry.addAttribute( 'uv', new THREE.BufferAttribute( new Float32Array( geometry.uvs ), 2 ) );
 
         }
 
-        // Create materials
-
-        var createdMaterials = [];
-
-        for (var mi = 0, miLen = materials.length; mi < miLen; mi++) {
-
-          var sourceMaterial = materials[mi];
-          var material = undefined;
-
-          if (this.materials !== null) {
-
-            material = this.materials.create(sourceMaterial.name);
+        var material;
 
-            // mtl etc. loaders probably can't create line materials correctly, copy properties to a line material.
-            if (isLine && material && !(material instanceof THREE.LineBasicMaterial)) {
+        if ( this.materials !== null ) {
 
-              var materialLine = new THREE.LineBasicMaterial();
-              materialLine.copy(material);
-              material = materialLine;
-
-            }
-
-          }
-
-          if (!material) {
-
-            material = (!isLine ? new THREE.MeshPhongMaterial() : new THREE.LineBasicMaterial());
-            material.name = sourceMaterial.name;
-
-          }
-
-          material.shading = sourceMaterial.smooth ? THREE.SmoothShading : THREE.FlatShading;
-
-          createdMaterials.push(material);
+          material = this.materials.create( object.material.name );
 
         }
 
-        // Create mesh
-
-        var mesh;
+        if ( !material ) {
 
-        if (createdMaterials.length > 1) {
+          material = new THREE.MeshPhongMaterial();
+          material.name = object.material.name;
 
-          for (var mi = 0, miLen = materials.length; mi < miLen; mi++) {
-
-            var sourceMaterial = materials[mi];
-            buffergeometry.addGroup(sourceMaterial.groupStart, sourceMaterial.groupCount, mi);
-
-          }
-
-          var multiMaterial = new THREE.MultiMaterial(createdMaterials);
-          mesh = (!isLine ? new THREE.Mesh(buffergeometry, multiMaterial) : new THREE.LineSegments(buffergeometry, multiMaterial));
-
-        } else {
-
-          mesh = (!isLine ? new THREE.Mesh(buffergeometry, createdMaterials[0]) : new THREE.LineSegments(buffergeometry, createdMaterials[0]));
         }
 
+        material.shading = object.material.smooth ? THREE.SmoothShading : THREE.FlatShading;
```

Comment:
`material.shading` has been removed.  `material.flatShading = true` should be used instead.

Reply:
Thats true..perhaps that could be done in another PR?

Reply:
@sohamkamani That makes sense. ~ I don't know the PR procedure outside of my work environment, new to the game 👍

Reply:
I sent a PR with just this fix https://github.com/sohamkamani/three-object-loader/pull/15

Reply:
merged 👍  Sorry for the (very) late reply